### ISSUE TITLE
allow custom file types

### ISSUE THREAD

**Issue by KiaraGrouwstra** [littleuniversestudios/angular-cli-tools#12]
So currently, there's a [fixed list](https://github.com/littleuniversestudios/angular-cli-tools/#custom-templates) of file types you can generate, all of which except html/style just seeming to add `<filetype>.ts` to the generated output.

I currently hoped to be able to generate Pug files instead of HTML. As it stands, changing the filetype to something not on the list appears to be unsupported; I appear to be getting a `fooundefined` file, whereas I'm hoping to make a `foo.component.pug` file or the like.

I'd wonder if there could be a nicer way to do things like this. e.g., if I could just put something like `.component.pug` for the file type, and have it just use that to decide the generated file names, then to me that'd be great (compared to the alternative of being limited to an upstream white-list). Heck, could it make sense to base output file names on input file names? Furthermore, might it be doable to have it just detect the files in a folder rather than manually listing them out?

I'll concede I'm ignorant with regard to the design process this library was founded on, but I'd be curious to learn more!

**Comment by nik-lus** [comment]
@tycho01 there was no design process at the outset, this was a weekend project that snowballed into a library. We just needed a quick way to generate common components/files in ng2 without being tied to the official ng2 CLI. Not my proudest code (there were no design considerations) but it does the job. 

As for .pug support (or any custom file support)...that's not on the radar any time soon. We've hit a stable version for now and built many ng2 projects with this so I'm not sure when the next release will be. 

My best suggestion is to fork it, poke around, and customize it for yourself. Apologies in advance, code can be hard to follow.

**Comment by KiaraGrouwstra** [comment]
Fair enough; thanks for your response! :)
